PPP password is now stored in this way in the snapshot.

MMaiero opened the pull request:
Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

MMaiero committed:
PPP password is now managed and stored as Password in the snapshot.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

MMaiero commented:
Close #1196

Commit:
Modified the getModemConfig() method to obtain password from properties as both <Password> and <String>.

Signed-off-by: ibinshtok <ilya.binshtok@eurotech.com>

ibinshtok commented:
@MMaiero I made one more change. Please take a quick look. If you agree with the change I'll merge this PR.

MMaiero committed:
Started refactoring of getModemConfig in NetworkConfiguration.

Minor cleanups in class.
Updated the tests to meet the new variables names.
Updated versions in distrib.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>
